### PR TITLE
Simplify contributing

### LINKED ISSUE #267: docs: simplify contributing page
### Overview

We need contributing docs more suitable for contributors of our project that should already know the basics. The main page should be kept simple with links to more detailed explanations. A short but useful contributor intro to the project will make it more accessible.

### Action Items

- [x] Create a `contributing` folder in `docs/` to hold contributing pages
- [x] Move most or all sections out of the contributing page into pages in `contributing/`
- [x] Split `CONTRIBUTING.md` from `docs/contributing/index.md`
  - [x] Replace links to `CONTRIBUTING.md` with short description and link to `contributing/index.md`
  - [x] Replace `contributing/index.md` sections with short descriptions and links to individual pages

### Instructions

- Keep the contributing page short but usable by experienced devs
- Link to detailed pages for clarification
- What this means is to expand the CONTRIBUTING.md file into a folder of contributing documents for everything related to PD development
    ```
    contributing
    ├── index.md
    ├── team.md
    ├── dev_environment.md
    ├── issues.md
    ├── git.md
    ├── documentation.md
    ├── ...
    ├── how-to
    │   ├── add-model-and-api-endpoints.md
    │   └── ...
    └── tools
        ├── docker.md
        └── ...
    ```

### PR DESCRIPTION
**Pull request by fyliu**
Fixes #267

### What changes did you make?

- Moved CONTRIBUTING.md sections into files and created links from the original file
  - The original file became the section index page.
- Created contributing/ directory and moved contributing files inside
- Moved tools/ and howto/ (Developer How-to Guides) into contributing/
- Updated links for the above changes.

### Why did you make the changes (we will use this info to test)?

- To split the long contributing docs file into topic pages so it'll be easier to add and make changes going forward.
- To organize contributing pages into the Contributing section. Leave main sections for user/customer documentation.

### Screenshots of Proposed Changes Of The Website

[Before link](https://hackforla.github.io/peopledepot/CONTRIBUTING/)
<details>
<summary>Visuals before changes are applied</summary>

- Boxes show where the pages were originally.
- The page table of contents used to be on the right.
- There were numbering in the titles, because the document was so long
  - Having numbers as part of the titles (rounded boxes) made it difficult to insert new content because of having to re-number everything after it.

2024-10-03 09 37 34 hackforla github io e4315503883e

</details>

[After link](https://fyliu.github.io/peopledepot/contributing/)
<details>
<summary>Visuals after changes are applied</summary>

- Now, all contributor pages are organized under the "Contributing" section (boxed), leaving the main sections for "customers" (API client developers).
- There's a Contributing index page with links pointing to the other pages.

2024-10-02 14 50 42 localhost 38543a596a76

- Now, page table of contents is shorter and reads better without numbering all the titles

2024-10-10 15 30 50 fyliu github io a1818e68d0a4

</details>

### CODE REVIEW

**Comment by fyliu** [comment]
Maybe the 2 vs. 3 column display needs some more thinking through. I'm going to remove that and recreate screenshots for this PR.

The reasoning is that the [Stripe docs](https://docs.stripe.com/sdks) use the 3-column format, and it's been the gold standard for software API documentation for a long time. Maybe it's something that users (client developers) prefer.

File: docs/contributing/dev_environment.md
Comment (shmonks) [inline docs/contributing/dev_environment.md:1]: Line 165: is the link to Work with Docker to be inserted later?

File: docs/contributing/howto/index.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# How-to Guides
+
+These are the developer guides for how to do specific things with the project.
+
+1. [Add model and API endpoints](add-model-and-api-endpoints.md)
+1. [Create initial data migrations](create-initial-data-migrations.md)
```
Comment (shmonks) [inline docs/contributing/howto/index.md:6]: Line 6: Change to 'Create initial data scripts' - and maybe change the name of the linked .md file too?

File: docs/contributing/dev_environment.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+1. Create a super user for logging into the web admin interface
+
+    ```bash
+    docker-compose exec web python manage.py createsuperuser --no-input
+    ```
+
+1. Browse to the web admin interface at `http://localhost:8000/admin/` and confirm the admin site is running. Use DJANGO_SUPERUSER_USERNAME and DJANGO_SUPERUSER_PASSWORD from .env.docker for credentials.
+
+See our documentation for [Working with Docker](http://localhost:8005/contributing/tools/docker/#working-with-docker) for more useful Docker commands.
```
Comment (shmonks) [inline docs/contributing/dev_environment.md:165]: Line 165: is the link to Working with Docker page to be inserted later?

File: docs/contributing/index.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Contributing
+
+Thank you for volunteering your time! The following is a set of guidelines for contributing to the peopledepot repository, which is hosted on GitHub.
+
+**Please make sure you have completed the onboarding process which includes joining the Hack for LA Slack, GitHub, and Google Drive. If you have not been onboarded, see the [Getting Started Page](https://www.hackforla.org/getting-started).** _Workshop attendees are granted a temporary exception from this requirement._
+
+- [Joining the team](team.md)
+
+- [Setting up the Developement Environment](dev_environment.md)
```
Comment (shmonks) [inline docs/contributing/index.md:9]: Line 9: change to 'Setting up the Development Environment'

File: docs/contributing/index.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Contributing
+
+Thank you for volunteering your time! The following is a set of guidelines for contributing to the peopledepot repository, which is hosted on GitHub.
```
Comment (shmonks) [inline docs/contributing/index.md:3]: Line 3: change to 'PeopleDepot repository'

File: docs/contributing/issues.md
Comment (shmonks) [inline docs/contributing/issues.md:1]: Updated link to Prioritized Backlog and corrected a typo in new commit

File: docs/contributing/tools/index.md
Comment (shmonks) [inline docs/contributing/tools/index.md:1]: Switched the order to match right-hand menu

File: docs/contributing/index.md (hunk 4 above)
Comment (shmonks) [reply docs/contributing/index.md]: I fixed it - but for some reason, it won't let me resolve this

**Review by shmonks — APPROVED** [review]
All good! A few tiny corrections and a couple of comments, that's all.

File: docs/contributing/howto/index.md (hunk 1 above)
Comment (fyliu) [reply docs/contributing/howto/index.md:6]: I don't think we should change the wording here.

Migrations is the term Django uses for this type of scripts. They don't work alone, but have to be run by Django's migration system. https://docs.djangoproject.com/en/5.1/topics/migrations/

It took a while to compare and figure out the best format to keep the initial data and insert them into the database. I should write up a DR to document why.

File: docs/contributing/dev_environment.md (hunk 2 above)
Comment (fyliu) [reply docs/contributing/dev_environment.md:165]: Great catch! localhost is definitely a placeholder link that didn't get cleaned up.

Comment (fyliu) [reply docs/contributing/dev_environment.md]: Fixed

File: docs/contributing/tools/index.md
Comment (fyliu) [reply docs/contributing/tools/index.md:1]: I un-pinned the one file that made the list not go in alphabetical order. It's slightly more maintainable this way.

**Pull request merged**